### PR TITLE
Optimze HTTP header access in test

### PR DESCRIPTION
**Pull request by themue**
**Description**

Card for related Checkmarx link describes possible resource exhaustion while looking for the `Ce-Type` header. 

This header is now addressed directly.

### CODE REVIEW

File: components/event-publisher-proxy/pkg/handler/beb/handler_test.go [hunk 1]
```diff
@@ -487,18 +487,16 @@ func validateApplicationName(appName string) testingutils.Validator {
...
-		for k, v := range r.Header {
-			if strings.ToLower(k) != strings.ToLower(testingutils.CeTypeHeader) {
-				continue
-			}
-			if strings.Contains(v[0], appName) {
-				return nil
-			}
-			return errors.New(fmt.Sprintf("expected the event-type [%s] to contain the application name [%s] for binary mode", v[0], appName))
+		// Default CE binary mode
+		value, ok := r.Header[testingutils.CeTypeHeader]
```
Comment (sayanh) [inline components/event-publisher-proxy/pkg/handler/beb/handler_test.go:493]: IMO r.Header.Get() fits well here as it takes of the case as well.

File: components/event-publisher-proxy/testing/utils.go [hunk 2]
```diff
@@ -21,10 +21,10 @@ import (
 
 const (
 	// binary cloudevent headers
-	CeIDHeader          = "CE-ID"
-	CeTypeHeader        = "CE-Type"
-	CeSourceHeader      = "CE-Source"
-	CeSpecVersionHeader = "CE-SpecVersion"
+	CeIDHeader          = "Ce-Id"
```
Comment (sayanh) [inline components/event-publisher-proxy/testing/utils.go:24]: Why is this needed as headers are case-insensitive. ?

File: components/event-publisher-proxy/pkg/handler/beb/handler_test.go (hunk 1 above)
Comment (themue) [reply components/event-publisher-proxy/pkg/handler/beb/handler_test.go:493]: Thought about the same but not sure about empty headers vs non-existing headers. `Get()` would return an empty string. But might be okay here for the test, so I'll change.

File: components/event-publisher-proxy/testing/utils.go (hunk 2 above)
Comment (themue) [reply components/event-publisher-proxy/testing/utils.go:24]: It's the Canonical MIME Header. But in case using `Get()` it's okay to set them to the lower-case ones as defined in the  standard. Here `Get()` changes them on the fly.

Comment (themue) [reply components/event-publisher-proxy/testing/utils.go]: Done.

**Review by sayanh — APPROVED** [review]
